### DISCUSSION TITLE
How to add computed property to model type?

### DISCUSSION THREAD

**Discussion by nilswsr** [prisma/prisma#20623] [Q&A]
### Question

I've added a computed property like [this](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#add-a-custom-field-to-query-results).

```javascript
import {PrismaClient} from "@prisma/client";

const prisma = new PrismaClient().$extends({
  result: {
    user: {
      fullName: {
        needs: { firstName: true, lastName: true },
        compute(user) {
          return `${user.firstName} ${user.lastName}`
        },
      },
    },
  },
})
```

When im trying to use my new property like the following, everything works fine:
```javascript
const user = prisma.user.findFirst({
   where: {
       firstName: "Anna"
   }
})
console.log(user.fullName)
```
The type is correctly recognized and there's no TypeError.

On the other hand, when using the imported `User`model type or the `GetPayload`-version, my new computed property isn't recognized. Both statements inside the `console.log()` don't work, because `fullName` is not known.
```javascript
import {Prisma} from "@prisma/client";
import {User} from "@prisma/client";

type UserWithPosts = Prisma.UserGetPayload<{
    include: {
        posts: true
    }
}>


function (user: User, userWithPosts: UserWithPosts) {
    console.log(user.fullName);
    console.log(userWithPosts.fullName);
}
```

Is there a way to solve this? Thanks in advance!

### Information about Prisma Schema, Client Queries and Environment (optional)

prisma                 : 5.0.0
@prisma/client    : 5.0.0

**Comment by ludralph** [comment]
Hi @nilswsr 👋 

Did you intend to use [`Prisma.Result`](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/type-utilities) in this case, rather than `GetPayload`?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
